Sources/Testing/Running/EntryPoint.swift

```diff
@@ -55,39 +60,127 @@ public func swiftPMEntryPoint() async -> Never {
   exit(exitCode)
 }
 
+// MARK: -
+
+/// List all of the given tests in the "specifier" format used by Swift Package
+/// Manager.
+///
+/// - Parameters:
+///   - tests: The tests to list.
+private func _listTestsForSwiftPM(_ tests: some Sequence<Test>) {
+  // Filter out hidden tests and test suites. Hidden tests should not generally
+  // be presented to the user, and suites (XCTestCase classes) are not included
+  // in the equivalent XCTest-based output.
+  let tests = tests.lazy
+    .filter { !$0.isSuite }
+    .filter { !$0.isHidden }
+
+  // Group tests by the name components of the tests' IDs. If the name
+  // components of two tests' IDs are ambiguous, present their source locations
+  // to disambiguate.
+  let allTestIDs = Dictionary(
+    grouping: tests.lazy.map(\.id),
+    by: \.nameComponents
+  ).values.lazy
+    .map { ($0, isAmbiguous: $0.count > 1) }
+    .flatMap { testIDs, isAmbiguous in
+      testIDs.lazy
+        .map { testID in
+          if !isAmbiguous, testID.sourceLocation != nil {
+            return testID.parent ?? testID
+          }
+          return testID
+        }
+    }.map(String.init(describing:))
+    .sorted(by: <)
+
+  // Print all the test IDs to the console in neutral sorted order.
+  for testID in allTestIDs {
+    print(testID)
+  }
+}
+
+/// Get an instance of ``Configuration`` given a sequence of command-line
+/// arguments passed from Swift Package Manager.
+///
+/// - Parameters:
+///   - args: The command-line arguments to interpret.
+///
+/// - Returns: An instance of ``Configuration``. Note that the caller is
+///   responsible for setting this instance's ``Configuration/eventHandler``
+///   property.
+///
+/// This function generally assumes that Swift Package Manager has already
+/// validated the passed arguments.
+private func _configurationForSwiftPMEntryPoint(withArguments args: [String]) -> Configuration {
```

Comment:
Can we get unit tests for this new logic?

Reply:
Yes, let me think a bit about what we'd want to test.